scripts/e2e/opencode-mcp-runtime.py

```
@@ -0,0 +1,317 @@
+EXPECTED_TOOLS = {"echo", "get-sum"}
+
+
+def build_pipelock(workdir: Path) -> Path:
+    """Compile pipelock from the surrounding worktree or honor PIPELOCK_BIN."""
+    bin_override = os.environ.get("PIPELOCK_BIN")
+    if bin_override:
+        print(f"Using pre-built pipelock at {bin_override}")
+        return Path(bin_override)
+    out = workdir / "pipelock"
+    print(f"Building pipelock from {REPO_ROOT} -> {out}")
+    subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)